contracts/WakuRln.sol

```diff
@@ -1,73 +1,95 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.15;
+pragma solidity 0.8.19;
 
-import {IPoseidonHasher} from "rln-contract/PoseidonHasher.sol";
 import {RlnBase, DuplicateIdCommitment, FullTree, InvalidIdCommitment} from "rln-contract/RlnBase.sol";
 import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
 
 error NotImplemented();
+error MalformedCommitmentToMessageLimitMap();
+
+function isValidCommitmentToMessageLimitMap(uint256[] calldata commitments, uint256[] calldata limits)
+    pure
+    returns (bool)
+{
+    uint256 commitmentsLen = commitments.length;
+    uint256 limitsLen = limits.length;
+    if (commitmentsLen != limitsLen) return false;
+    return true;
+}
 
 contract WakuRln is Ownable, RlnBase {
     uint16 public immutable contractIndex;
 
-    constructor(
-        address _poseidonHasher,
-        uint16 _contractIndex
-    ) Ownable() RlnBase(0, 20, _poseidonHasher, address(0)) {
+    constructor(uint256 _maxMessageLimit, uint16 _contractIndex)
+        Ownable()
+        RlnBase(0, 20, _maxMessageLimit, address(0))
+    {
         contractIndex = _contractIndex;
     }
 
+    modifier onlyValidCommitmentToMessageLimitMap(uint256[] calldata commitments, uint256[] calldata limits) {
+        if (!isValidCommitmentToMessageLimitMap(commitments, limits)) {
+            revert MalformedCommitmentToMessageLimitMap();
+        }
+        _;
+    }
+
     /// Registers a member
     /// @param idCommitment The idCommitment of the member
-    function _register(uint256 idCommitment) internal {
-        _validateRegistration(idCommitment);
+    /// @param userMessageLimit The userMessageLimit of the member
+    function _register(uint256 idCommitment, uint256 userMessageLimit) internal {
+        _validateRegistration(idCommitment, userMessageLimit);
 
         members[idCommitment] = idCommitmentIndex;
+        indexToCommitment[idCommitmentIndex] = idCommitment;
         memberExists[idCommitment] = true;
+        userMessageLimits[idCommitment] = userMessageLimit;
 
-        emit MemberRegistered(idCommitment, idCommitmentIndex);
+        emit MemberRegistered(idCommitment, userMessageLimit, idCommitmentIndex);
         idCommitmentIndex += 1;
     }
 
-    function register(uint256[] calldata idCommitments) external onlyOwner {
-        uint256 len = idCommitments.length;
-        for (uint256 i = 0; i < len; ) {
-            _register(idCommitments[i]);
+    function register(uint256[] calldata commitments, uint256[] calldata limits)
```

Comment:
perhaps nitpick:
`register(uint256[] calldata idCommitments, uint256[] calldata userMessageLimits)`

instead of `commitments`, `limits`

so its easier to see that it maps to
`register(uint256 idCommitment, uint256 userMessageLimit)`

* idCommitment -> idCommitments
* userMessageLimit -> userMessageLimits

Reply:
will make that change 👍